### ISSUE TITLE
Not using the compiler options for C++ when using C++14 syntax 

### ISSUE THREAD

**Issue by ricardopieper** [hebaishi/linter-gcc#25]
So, after embarassing myself over here https://github.com/AtomLinter/linter-cpplint/issues/38, I hope i'm doing the right thing this time, i'll copy and paste from that.

I have a package called language-cpp14 installed. I just installed because it sounds cool and seems to add some extra neat things, but I don't really know if I need it. I'm new to C++, so sorry if I'm not making any sense.

I have a perfectly valid C++ program using C++11 stuff. But since i've installed the language-cpp14 extension, the current language i'm using in Atom is also called C++14.

I know exactly where the problem lies: main.js, line 100:

    if(grammar_name === "C++") {
        flags = settings.gccDefaultCppFlags;
    } else if(grammar_name === "C") {
       flags = settings.gccDefaultCFlags;
    }

I figured that grammar_name is C++14 so it doesn't use the flags. And throws over 5000 errors when I save the file, because it seems that G++ will default to the old c++.

I'm OK with using the normal C++ though, and that's what I'll do, probably. I was just wondering if that's a problem. Would that be correct to check whether grammar_name contains the string c++14?

**Comment by hebaishi** [comment]
Thank you for opening this issue. This is an interesting problem, and I will make sure it is fixed in the next release of linter-gcc. I am currently working on a big update to linter-gcc that will add linting on-the-fly so I will add this to my to-do list.

**Comment by hebaishi** [comment]
I have made the required change. Linting C++14 (both on-save and on-the-fly) should now be working.

**Comment by hebaishi** [comment]
Hello. Just wanted to confirm that it's working for you now.

**Comment by ricardopieper** [comment]
Yo!

I'm having some internet troubles at home, so i've not been able to test it yet...as soon as I solve that, i'll test it ;D

**Comment by hebaishi** [comment]
Hello. Any update on your C++14 issue?

**Comment by ricardopieper** [comment]
Hi, I updated the package and apparently the problem doesn't happen anymore, but I saw the code, and you're gonna need another fix for C++17 I guess :p but that's somewhat far away from our reality.

Isn't it better to check if the string starts with "C++" or something like that instead? Because, as far as I understand, you check if the selected language is C++, and if it is, then you just invoke some process, right?

But anyway, the problem doesn't happen anymore :D

**Comment by hebaishi** [comment]
Yes that is a good point. I will change the code so it looks for ```C++``` somewhere in the grammar name. It should work with all ```C++``` grammars then. Let's keep this issue open until I've made the change.

**Comment by hebaishi** [comment]
Ok. Just published the new version. The change has been made: 930d4398afef7f74b4cfc3d0b2dbb32cece527be. Could you update and confirm it's working for you with C++14?

**Comment by ricardopieper** [comment]
Yes, it's working nicely :P
